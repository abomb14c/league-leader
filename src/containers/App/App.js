import React, { Component } from 'react';
import './App.css';
import { Route, Redirect, Switch, withRouter } from 'react-router-dom';
import { connect } from 'react-redux'
import Navigation from '../../components/Navigation/Navigation';
import Leagues from '../../components/Leagues/Leagues';
import Login from '../../components/Login/Login';
import Home from '../Home/Home';
import { fetchEnglandScores } from '../../apiCalls/apiCalls';
import {updateEnglishSoccer} from '../../actions/handleSoccer/handleSoccer';

class App extends Component {

  getSoccerData = async () => {
    const soccerStats = await fetchEnglandScores();
    await this.props.handleEnglishSoccer(soccerStats)
  }

  componentDidMount = () => {
    this.getSoccerData();
  }

  render() {
    return (
      <div className="App">
        <header>
          <Navigation />
        </header>
        <Switch>
        <Route
          exact path="/login"
          render={() => (
            this.props.user.user_id ?
              <Redirect to="/" /> :
              <Login />
          )}
        />
        <Route
          exact path="/leagues"
          component={Leagues} 
        />
        <Route
          exact path="/"
          component={Home}
        />
        </Switch>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  user: state.user
})

export const mapDispatchToProps = dispatch => ({
  handleEnglishSoccer: (englishSoccerStats) => 
  dispatch(updateEnglishSoccer(englishSoccerStats))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
