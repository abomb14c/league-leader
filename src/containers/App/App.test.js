import React from 'react';
import ReactDOM from 'react-dom';
import { App, mapDispatchToProps, mapStateToProps } from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  let mockHandleEnglishSoccer;
  let mockHandleNba;

  beforeEach(() => {
    mockHandleEnglishSoccer = jest.fn();
    mockHandleNba = jest.fn();

    wrapper = shallow(<App 
      handleEnglishSoccer={mockHandleEnglishSoccer}
      handleNba={mockHandleNba}
    />, { disableLifecycleMethods: true });
  });
  
  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });
});