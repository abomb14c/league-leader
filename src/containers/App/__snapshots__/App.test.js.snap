// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    handleEnglishSoccer={[Function]}
    handleNba={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <Connect(Component) />
</header>,
        <Switch>
          <Route
                    exact={true}
                    path="/login"
                    render={[Function]}
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/leagues"
          />
          <Route
                    exact={true}
                    path="/"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/setupleague"
                    render={[Function]}
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/draft"
          />
</Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Component) />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/login"
              render={[Function]}
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/leagues"
/>,
            <Route
              exact={true}
              path="/"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/setupleague"
              render={[Function]}
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/draft"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/login",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/leagues",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/setupleague",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/draft",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <Connect(Component) />
</header>,
          <Switch>
            <Route
                        exact={true}
                        path="/login"
                        render={[Function]}
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/leagues"
            />
            <Route
                        exact={true}
                        path="/"
                        render={[Function]}
            />
            <Route
                        exact={true}
                        path="/setupleague"
                        render={[Function]}
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/draft"
            />
</Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Component) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/login"
                render={[Function]}
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/leagues"
/>,
              <Route
                exact={true}
                path="/"
                render={[Function]}
/>,
              <Route
                exact={true}
                path="/setupleague"
                render={[Function]}
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/draft"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/login",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/leagues",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/setupleague",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/draft",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
